ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -695,25 +710,116 @@ private Dataset executeGetDatasetStatement(PreparedStatement statement, int acco
         retentionCount, userTags);
   }
 
+  /**
+   * Get version schema corresponding to the dataset.
+   * @param statement the mysql statement to get the version schema of the dataset.
+   * @param accountId the id of the parent account.
+   * @param containerId the if of the container.
+   * @param datasetName the name of the dataset.
+   * @return the name of the dataset.
+   * @throws SQLException
+   */
+  private Dataset.VersionSchema executeGetVersionSchema(PreparedStatement statement, int accountId, int containerId,
+      String datasetName) throws SQLException {
+    ResultSet resultSet = null;
+    Dataset.VersionSchema versionSchema;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      resultSet = statement.executeQuery();
+      if (!resultSet.next()) {
+        throw new SQLException(
+            "Version Schema not found for account: " + accountId + " container: " + containerId + " dataset: "
+                + datasetName);
+      }
+      versionSchema = Dataset.VersionSchema.values()[resultSet.getInt(VERSION_SCHEMA)];
+    } finally {
+      //If result set is not created in a try-with-resources block, it needs to be closed in a finally block.
+      closeQuietly(resultSet);
+    }
+    return versionSchema;
+  }
+
+  /**
+   * Convert string version to long value.
+   * @param version The string version.
+   * @param versionSchema The {@link com.github.ambry.account.Dataset.VersionSchema} of the dataset.
+   * @return the long value of version.
+   */
+  private long getVersionBasedOnSchema(String version, Dataset.VersionSchema versionSchema) {
+    switch (versionSchema) {
+      case TIMESTAMP:
+      case MONOTONIC:
+        return Long.parseLong(version);
```

Comment:
should check if the version is not numeric value here as well.

Reply:
Fixed it.